### ISSUE TITLE
Add error checking for user config

### ISSUE THREAD

**Issue by mehalter** [kabinspace/AstroVim#232]
We should add some form of error checking for user configuration with some form of notification/warning on compilation error.

I've had some ideas for this, planning on implementing tomorrow.

**Comment by AXGKl** [comment]
@mehalter : While you are at that a note:

Was creating an *unattended installer* for Astrovim, with personal user settings from our internal repos (e.g. for cloud servers in DEV, i.e. where you want to develop on).

Question: Can you make the error checking so that unattended installers do install the wanted user plugins, even IF there are failures in `polish()`, where users require them?  `pcall` checks for any action on a plugin are often forgotten, people tend to just copy the install instructions from github, e.g. https://github.com/uga-rosa/cmp-dictionary.

Details:

I run the installer like this:

```bash
nvim  --headless -c 'autocmd User PackerComplete quitall' -c 'PackerSync'
```

In order to have it install the user plugins in any case I currently have to do this before running: `export setup_mode=true` and tell all users with personal configs to structure their configs like this:



```lua
local user_plugins = {
	"ThePrimeagen/refactoring.nvim",
	....
}

-------------    EXIT EARLY IN SETUP MODE:
if os.getenv("setup_mode") then
	return { plugins = { init = user_plugins } }
end
-------------    

local config = {
        plugins = { init = user_plugins, ...}
}
return config
``` 

so that the unattended install runs through.

**Comment by mehalter** [comment]
https://asciinema.org/a/zchfnljtyXwKWYI5jjrrw8f4V

Here is what it's looking like. Basically if the file exists and it's unable to do the `require` then it will show the error while proceeding without loading the file. This does have limitations because stuff like functions are lazy loaded so they won't necessarily trigger this, but I'm not sure of a better way of solving that at the moment. This gets pretty close though

**Comment by mehalter** [comment]
@AXGKl I think that this adds some unnecessary complexity to the `init.lua` file that will make the base user config harder to read on initial look as well as require the user to set that environment variable to get the feature. This is available though if you split up your `init.lua` into separate files as described in the wiki (https://github.com/kabinspace/AstroVim/wiki/Advanced-Configuration#splitting-up-configuration-to-multiple-files)

Basically if the files are separate then they will be loaded separately and if one breaks the others can still load just fine 😄  also makes the user config a lot easier to maintain (in my opinion). If you want to see an example of a fully separated out `user/` folder you can check out my configuration repository (https://git.mehalter.com/mehalter/AstroVim_user/) you can go as far as to not having a `user/init.lua` file at all 👀

**Comment by AXGKl** [comment]
>  my configuration repository (https://git.mehalter.com/mehalter/AstroVim_user/) 

Oh man, that repo is worth gold. So much information extractable....  And yes, that would autoinstall just fine, w/o that early exit hack.

=> Strongly suggest to link to it in the Readme and also maybe to other AV blackbelts' configs.

As a noob regarding how the framework ticks and just a bit of lua, you only have that one `user_example` - and that's not that much information, when one really wants to tweak stuff.
